packages/flutter/lib/src/painting/debug.dart

```diff
@@ -31,3 +33,19 @@ bool debugAssertAllPaintingVarsUnset(String reason, { bool debugDisableShadowsOv
   }());
   return true;
 }
+
+/// Signature for a method that returns an [HttpClient].
+///
+/// Used by [debugNetworkImageHttpClientProvider].
+typedef HttpClientProvider = HttpClient Function();
+
+/// Provider from which [NetworkImage] will get its [HttpClient] in debug builds.
+///
+/// If this value is unset, [NetworkImage] will use its own internally-managed
+/// [HttpClient].
+///
+/// This setting can be overridden for testing to ensure that each test receives
+/// a mock client that hasn't been affected by other tests.
+///
+/// This value is ignored in non-debug builds.
+HttpClientProvider debugNetworkImageHttpClientProvider;
```

Comment:
Should `debugAssertAllPaintingVarsUnset` also check that this is unset?

Reply:
Yep - done.